### PR TITLE
Remove assumption that there can be only one root from IntegrationPoint

### PR DESCRIPTION
**Pull request by KovalevAndrey**
## Description

Fixes https://github.com/bumble-tech/appyx/issues/81

1. Removed assumption that only one Appyx root can exist
2. IntegrationPoint should be attached at the moment of the node creation. This way we can guarantee that a `Node` has `IntegrationPoint` before `node.Compose()` is called

## Check list

- [x] I have updated `CHANGELOG.md` if required.
- [x] I have updated documentation if required.

### CODE REVIEW

File: core/src/main/java/com/bumble/appyx/core/integration/NodeHost.kt [hunk 1]
```diff
@@ -29,9 +30,8 @@ fun <N : Node> NodeHost(
     factory: NodeFactory<N>
 ) {
     val node by rememberNode(factory, customisations)
-    DisposableEffect(integrationPoint) {
-        integrationPoint.attach(node)
-        onDispose { integrationPoint.detach() }
+    LaunchedEffect(integrationPoint, node) {
+        node.integrationPoint = integrationPoint
```
Comment (LachlanMcKee) [inline core/src/main/java/com/bumble/appyx/core/integration/NodeHost.kt:34]: Does this need to be in a launched effect?

Comment (KovalevAndrey) [reply core/src/main/java/com/bumble/appyx/core/integration/NodeHost.kt:34]: yes, because otherwise it will be called on every recomposition

Comment (LachlanMcKee) [reply core/src/main/java/com/bumble/appyx/core/integration/NodeHost.kt:34]: Good point

**Review by LachlanMcKee — APPROVED** [review]

File: core/src/main/java/com/bumble/appyx/core/integration/NodeHost.kt [hunk 2]
```diff
@@ -28,11 +29,7 @@ fun <N : Node> NodeHost(
     integrationPoint: IntegrationPoint,
     factory: NodeFactory<N>
 ) {
-    val node by rememberNode(factory, customisations)
-    DisposableEffect(integrationPoint) {
-        integrationPoint.attach(node)
-        onDispose { integrationPoint.detach() }
-    }
+    val node by rememberNode(factory, customisations, integrationPoint)
```
Comment (CherryPerry) [inline core/src/main/java/com/bumble/appyx/core/integration/NodeHost.kt:32]: `NodeCustomisationDirectory` is neither Stable nor Immutable, does it affects us?
Also if I do not pass `customisations` we create new `NodeCustomisationDirectoryImpl` on each composition.
(can be addressed as a separate PR)

**Review by CherryPerry — APPROVED** [review]

File: core/src/main/java/com/bumble/appyx/core/integration/NodeHost.kt (hunk 2 above)
Comment (KovalevAndrey) [reply core/src/main/java/com/bumble/appyx/core/integration/NodeHost.kt:32]: good point, we can remember it to prevent unnecessary recompositions

**Pull request merged**